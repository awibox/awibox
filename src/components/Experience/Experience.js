import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
// Styles
import styles from './Experience.scss';

const Experience = (props) => (
  <div className={cn(styles.experience, 'row')}>
    {props.data.map((item) => (
      <div key={item.name} className="col-md-6">
        <div className="skill clearfix">
          <h4>{item.name}</h4>
          <div className={styles.skillValue}>{item.years} years</div>
        </div>
        <div className={styles.skillContainer}>
          <div className={styles.skillPercentage}></div>
        </div>
      </div>
    ))}
  </div>
);

Experience.propTypes = {
  data: ImmutablePropTypes.list,
  className: PropTypes.string,
};

export default Experience;
