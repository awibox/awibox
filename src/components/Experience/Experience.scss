.experience {
  margin-bottom: 30px;
  h4 {
    font-size: 13px;
    line-height: 1.1em;
    position: relative;
    float: left;
    margin: 0 0 4px;
  }
  .skillContainer {
    position: relative;
    display: inline-block;
    background-color: #fff;
    border: 1px solid #fea621;
    -moz-border-radius: 8px;
    -webkit-border-radius: 8px;
    border-radius: 8px;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    height: 10px;
    margin-bottom: 8px;
    width: 100%;
    &:before {
      content: '';
      position: absolute;
      background-color: #fea621;
    }
  }
  .skillValue {
    font-size: 11px;
    line-height: 1.1em;
    position: relative;
    float: right;
    margin: 0 0 4px;
    color: #aaa;
  }
  .skillPercentage {
    background-color: #fea621;
    border: 2px solid #fff;
    -moz-border-radius: 9px;
    -webkit-border-radius: 9px;
    border-radius: 9px;
    height: 8px;
    padding: 0;
  }
}
