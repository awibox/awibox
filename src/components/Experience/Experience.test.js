import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';
import Experience from './Experience';

const mockSkills = List([
  {
    name: 'React',
    years: 4,
    percentage: 85,
  },
  {
    name: 'Angular',
    years: 7,
    percentage: 75,
  },
  {
    name: 'TypeScript',
    years: 6,
    percentage: 70,
  },
]);

describe('Experience', () => {
  it('should render correctly', () => {
    const component = shallow(<Experience data={mockSkills} />);
    expect(component).toMatchSnapshot();
  });
});
