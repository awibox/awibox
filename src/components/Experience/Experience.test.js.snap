// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience should render correctly 1`] = `
<div
  className="experience row"
>
  <div
    className="col-md-6"
    key="React"
  >
    <div
      className="skill clearfix"
    >
      <h4>
        React
      </h4>
      <div
        className="skillValue"
      >
        4
         
        years
      </div>
    </div>
    <div
      className="skillContainer"
    >
      <div
        className="skillPercentage"
        style={
          Object {
            "width": "85%",
          }
        }
      />
    </div>
  </div>
  <div
    className="col-md-6"
    key="Angular"
  >
    <div
      className="skill clearfix"
    >
      <h4>
        Angular
      </h4>
      <div
        className="skillValue"
      >
        7
         
        years
      </div>
    </div>
    <div
      className="skillContainer"
    >
      <div
        className="skillPercentage"
        style={
          Object {
            "width": "75%",
          }
        }
      />
    </div>
  </div>
  <div
    className="col-md-6"
    key="TypeScript"
  >
    <div
      className="skill clearfix"
    >
      <h4>
        TypeScript
      </h4>
      <div
        className="skillValue"
      >
        6
         
        years
      </div>
    </div>
    <div
      className="skillContainer"
    >
      <div
        className="skillPercentage"
        style={
          Object {
            "width": "70%",
          }
        }
      />
    </div>
  </div>
</div>
`;
